### PR TITLE
CCM-5828: Added details on how to generate hmac signature

### PR DESCRIPTION
**Pull request by TheWorldWideWebb**
## Summary
Added how to verify hmac signature in the docs.

Text added: 
This can be achieved by passing in the body crossed with the applicationID and apikey e.g. `hmacsha256(body, 'applicationId.apiKey')

Text removed:
Details on this will be provided during the onboarding process.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: specification/callbacks/channel-subscription-openapi-spec.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "/nhs-notify-callbacks/v1/channel-status": {
```
Comment (simonlabarere) [inline specification/callbacks/channel-subscription-openapi-spec.json:14]: Do we need to update the doc for message status callbacks as well?

**Review by simonlabarere — APPROVED** [review]

File: specification/callbacks/channel-subscription-openapi-spec.json (hunk 1 above)
Comment (TheWorldWideWebb) [reply specification/callbacks/channel-subscription-openapi-spec.json:14]: Its actually not in this one as all, the description is alot less descriptive in general.

File: specification/callbacks/channel-subscription-openapi-spec.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
...
...st with an invalid signature you should ignore it and respond with a `403 Forbidden`.\n\nEvery request includes an idempotencyKey located in the meta collection of the body. This can help ensure your system remains idempotent, capable of managing duplicate delivery of callbacks. It's important to note that requests may be delivered non-sequentially.\n\nIf a request fails, our retry policy will make up to three attempts with intervals of five seconds between each attempt.\n\nThe default behaviour of NHS Notify will make a callback to this endpoint when:\n\n* the message has been delivered (via any channel)\n* the message could not be delivered to a given channel (but may be deliverable by an alternative channel)\n* the message could not be delivered by any channel\n\nCallbacks can be received for additional state transitions subject to the needs of the user. These additional state transitions can be requested during onboarding. These statuses are:\n\n* `pending_enrichment` - the message is currently pending enrichment\n* `enriched` - we have queried PDS for this patient's details and now know how to contact this individual\n* `sending` - the message is in the process of being sent",
```
Comment (aaron-downing) [inline specification/callbacks/channel-subscription-openapi-spec.json:17]: I realise this was here before but maybe reword the following:
 ```You will need to validate the signature to verify the response has come from an authorized sender```
to
```You will need to validate the signature to verify the response has come from NHS Notify```

File: specification/documentation/APIDescription.md [hunk 3]
```diff
@@ -161,7 +161,7 @@ We will send your API key in the `x-api-key header`. Your service should respond
 * `401 Unauthorized` if the API key is not received
 * `401 Unauthorized` if the API key is invalid
 
-We will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from an authorized sender. Details on this will be provided during the onboarding process. If you receive a request with an invalid signature you should ignore it and respond with a `403 Forbidden`.
+We will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from an authorized sender. This can be achieved by passing in the body crossed with the applicationID and apikey e.g. `hmacsha256(body, 'applicationId.apiKey')`. If you receive a request with an invalid signature you should ignore it and respond with a `403 Forbidden`.
```
Comment (aaron-downing) [inline specification/documentation/APIDescription.md:164]: I worry `hmacsha256(body, 'applicationId.apiKey')` may be slightly confusing.

Possibly something like `This can be achieved by hashing the request body using the HMAC-SHA256 algorithm with a secret value that is comprised of a concatenation of your APIM application ID and the API key that we provide you. The secret takes the following form <APPLICATION_ID>.<API_KEY>`

Comment (TheWorldWideWebb) [reply specification/documentation/APIDescription.md:164]: Atleast make it a suggestion 😆 
```suggestion
This can be achieved by hashing the request body using the HMAC-SHA256 algorithm with a secret value that is comprised of a concatenation of your APIM application ID and the API key that we provide you. The secret takes the following form '<APPLICATION_ID>.<API_KEY>'.
```

File: specification/documentation/APIDescription.md [hunk 4]
```diff
@@ -161,7 +161,8 @@ We will send your API key in the `x-api-key header`. Your service should respond
 * `401 Unauthorized` if the API key is not received
 * `401 Unauthorized` if the API key is invalid
 
-We will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from an authorized sender. Details on this will be provided during the onboarding process. If you receive a request with an invalid signature you should ignore it and respond with a `403 Forbidden`.
+We will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from an NHS Notify.
```
Comment (aaron-downing) [inline specification/documentation/APIDescription.md:164]:
```suggestion
We will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from NHS Notify.
```

Comment (aaron-downing) [reply specification/documentation/APIDescription.md:164]: Better? 😉

File: specification/callbacks/channel-subscription-openapi-spec.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
...
```
Comment (TheWorldWideWebb) [inline specification/callbacks/channel-subscription-openapi-spec.json:17]:
```suggestion
        "description": "You may develop this endpoint on your service if you want to receive callbacks from NHS Notify.\n\nWe have created an OpenAPI specification detailing the behaviour of the endpoint that consumers should create to subscribe to callbacks.\n\nWe will send your API key in the `x-api-key` header. Your service should respond with:\n\n* `401 Unauthorized` if the API key is not received\n* `401 Unauthorized` if the API key is invalid\n\nWe will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from NHS Notify. This can be achieved by hashing the request body using the HMAC-SHA256 algorithm with a secret value that is comprised of a concatenation of your APIM application ID and the API key that we provide you. The secret takes the following form '<APPLICATION_ID>.<API_KEY>'. If you receive a request with an invalid signature you should ignore it and respond with a `403 Forbidden`.\n\nEvery request includes an idempotencyKey located in the meta collection of the body. This can help ensure your system remains idempotent, capable of managing duplicate delivery of callbacks. It's important to note that requests may be delivered non-sequentially.\n\nIf a request fails, our retry policy will make up to three attempts with intervals of five seconds between each attempt.\n\nThe default behaviour of NHS Notify will make a callback to this endpoint when:\n\n* the message has been delivered (via any channel)\n* the message could not be delivered to a given channel (but may be deliverable by an alternative channel)\n* the message could not be delivered by any channel\n\nCallbacks can be received for additional state transitions subject to the needs of the user. These additional state transitions can be requested during onboarding. These statuses are:\n\n* `pending_enrichment` - the message is currently pending enrichment\n* `enriched` - we have queried PDS for this patient's details and now know how to contact this individual\n* `sending` - the message is in the process of being sent",
```
```suggestion
        "description": "You may develop this endpoint on your service if you want to receive callbacks from NHS Notify.\n\nWe have created an OpenAPI specification detailing the behaviour of the endpoint that consumers should create to subscribe to callbacks.\n\nWe will send your API key in the `x-api-key` header. Your service should respond with:\n\n* `401 Unauthorized` if the API key is not received\n* `401 Unauthorized` if the API key is invalid\n\nWe will send you a HMAC-SHA256 signature in the `x-hmac-sha256-signature` header. You will need to validate the signature to verify the response has come from an NHS Notify. This can be achieved by hashing the request body using the HMAC-SHA256 algorithm with a secret value that is comprised of a concatenation of your APIM application ID and the API key that we provide you. The secret takes the following form '<APPLICATION_ID>.<API_KEY>'. If you receive a request with an invalid signature you should ignore it and respond with a `403 Forbidden`.\n\nEvery request includes an idempotencyKey located in the meta collection of the body. This can help ensure your system remains idempotent, capable of managing duplicate delivery of callbacks. It's important to note that requests may be delivered non-sequentially.\n\nIf a request fails, our retry policy will make up to three attempts with intervals of five seconds between each attempt.\n\nThe default behaviour of NHS Notify will make a callback to this endpoint when:\n\n* the message has been delivered (via any channel)\n* the message could not be delivered to a given channel (but may be deliverable by an alternative channel)\n* the message could not be delivered by any channel\n\nCallbacks can be received for additional state transitions subject to the needs of the user. These additional state transitions can be requested during onboarding. These statuses are:\n\n* `pending_enrichment` - the message is currently pending enrichment\n* `enriched` - we have queried PDS for this patient's details and now know how to contact this individual\n* `sending` - the message is in the process of being sent",
```
